File: 344_Reverse String.js
/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
var reverseString = function(s) {
    const halfSize = Math.floor(s.length / 2);
    const size = s.length;
    for(let i = 0; i < halfSize; ++i) {
        let temp = s[i];
        s[i] = s[size - 1 - i];
        s[size - 1 - i] = temp;
    }
};

/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
var reverseString = function(s) {
    
    const size = s.length;
    const halfSize = Math.floor(size / 2);
    
    for(let i = 0; i < halfSize; ++i) {
        let temp = s[i];
        s[i] = s[size - 1 - i];
        s[size - 1 - i] = temp;
    }
};

File: 344_Reverse String.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
        const int size = s.size();
        const int half_size = size >> 1;
        
        for(int i = 0; i < half_size; ++i) {
            char temp = s[i];
            s[i] = s[size - 1 - i];
            s[size - 1 - i] = temp;
        }
    }
};

class Solution {
public:
    void reverseString(vector<char>& s) {
        int start = 0, end = s.size() - 1;
        char ch = 0;
        
        while(start < end) {
            ch = s[start];
            s[start] = s[end];
            s[end] = ch;
            
            ++start;
            --end;
        }
    }
};